src/conn_state.c

```diff
@@ -102,6 +120,10 @@ enftun_conn_state_prepare(struct enftun_conn_state* conn_state,
 int
 enftun_conn_state_close(struct enftun_conn_state* conn_state)
 {
+    int rc = enftun_heartbeat_free(&conn_state->heartbeat);
+    if (0 != rc)
```

Comment:
Still need to close netlink, even if the hearbeat free fails. Just log a warning and continue.